com.unity.multiplayer.mlapi/Runtime/Core/SnapshotSystem.cs

```diff
@@ -275,9 +273,20 @@ public class SnapshotSystem : INetworkUpdateSystem, IDisposable
         private NetworkManager m_NetworkManager = NetworkManager.Singleton;
         private Snapshot m_Snapshot = new Snapshot(NetworkManager.Singleton, false);
         private Dictionary<ulong, Snapshot> m_ClientReceivedSnapshot = new Dictionary<ulong, Snapshot>();
+        private Dictionary<ulong, ClientRtt> m_ClientRtts = new Dictionary<ulong, ClientRtt>();
 
         private ushort m_CurrentTick = 0;
 
+        internal ClientRtt Rtt(ulong clientId)
```

Comment:
we don't call `NotifySend` or `NotifyAck` from anywhere yet.
are you planning to have RTT & SnapshotSystem integration in this PR too?

Reply:
sorry, just saw this:

> For now, those values are not filled, they will be in a future PR. But it should allow other development to proceed as it sets the API they will use.

AFAIK, we primarily wanted to have RTT for snapshots and NetVars. are there any other systems being unblocked by this PR? If not, personally, I think RTT integration with snapshot should go under one single PR — otherwise, we're not really shipping this as a feature but as a boilerplate. what would you think?

Reply:
Yes, there is some other systems blocked by this this. Luke's work on Time System will integrate with this.

Reply:
alright. just double-checking: without having this functioning (yet) but still having its boilerplate in-place would unblock Luke because his work on TimeSystem is not dependant to RTT calculation but merely its existence? I'm confused. @LukeStampfli ? 👀

Reply:
So based on your feedback on the time system I got, I'm exploring using the existing time sync for the time system until this is ready @MFatihMAR. I'm not sure yet whether that will work well and for release we will want to use Jeff's system. knowing the API with which I will integrate with makes it easier for me to plan ahead.

But yeah just an API which doesnt' work yet does not "unblock" me but it allows me to push my work with the confidence that there is a high probability that I will be able to integrate the time system in the future.

Reply:
so, having this "agreed in principle" would unblock you right? if so, again, I think this PR should go in with its snapshot integration/implementation in a functioning state, otherwise we're shipping a boilerplate, not a feature. I do not expect public APIs to change at this point, Luke is already unblocked from that standpoint then. what would you think @jeffreyrainy ? 👀

Reply:
My position is unchanged. I'd still like to PR this. Thanks for raising your concern.

Reply:
sure, fair enough.